### PR TITLE
Notifications GQL schema

### PR DESCRIPTION
**Pull request by CarsonF**
https://seed-company-squad.monday.com/boards/5989610236/pulses/7542832281

### CODE REVIEW

File: src/components/notifications/dto/notification.dto.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  implements: [Resource],
+})
+export class Notification extends Resource {
+  static readonly Props = keysOf<Notification>();
+  static readonly SecuredProps = keysOf<SecuredProps<Notification>>();
+
+  readonly owner: LinkTo<'User'>;
+
+  @Field(() => Boolean)
+  readonly unread: boolean;
```
Comment (CarsonF) [inline src/components/notifications/dto/notification.dto.ts:17]: Or would the flipped `read` be better?

File: src/components/notifications/notification.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  @Query(() => NotificationList)
+  async notifications(
+    @LoggedInSession() session: Session,
+    @ListArg(NotificationListInput) input: NotificationListInput,
+  ): Promise<NotificationList> {
+    throw new NotImplementedException().with(input, session);
+  }
+
+  @Mutation(() => Notification)
+  async markNotificationRead(
```
Comment (CarsonF) [inline src/components/notifications/notification.resolver.ts:26]: Actually I think 
```suggestion
  async readNotification(
```
would be better.

File: src/components/notifications/dto/notification.dto.ts (hunk 1 above)
Comment (bryanjnelson) [reply src/components/notifications/dto/notification.dto.ts:17]: That's a tough one; I was thinking about that before I saw your comment.  Typically I would say to flip it; positive approach over double-negative and all that...but this case is so well known and little thought of as "Oh, these are my 'read' notifications."  Of course, that's from a UI/visual perspective, so maybe that shouldn't play in to the model.  Kinda torn here...just typing it out so far.

File: src/components/notifications/notification.resolver.ts (hunk 2 above)
Comment (bryanjnelson) [reply src/components/notifications/notification.resolver.ts:26]: Eh, the problem with `readNotification` is it's ambiguous.   Is "read" a past tense verb, a present tense verb, or a noun here without context?

Comment (CarsonF) [reply src/components/notifications/notification.resolver.ts:26]: True. But this is in the context of a mutation action?

Comment (bryanjnelson) [reply src/components/notifications/notification.resolver.ts:26]: Point.  I can get on board with it.

File: src/components/notifications/dto/notification.dto.ts (hunk 1 above)
Comment (rdonigian) [reply src/components/notifications/dto/notification.dto.ts:17]: Put me down as another one who's torn the name, but I would keep it as is.

File: src/components/notifications/dto/notification.dto.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+export class SimpleTextNotification extends Notification {
+  @Field(() => String)
+  readonly content: string;
+}
+
+declare module '~/core/resources/map' {
+  interface ResourceMap {
+    Notification: typeof Notification;
+  }
+  // interface ResourceDBMap {
```
Comment (rdonigian) [inline src/components/notifications/dto/notification.dto.ts:32]: Is there a need for the comment here?

Comment (CarsonF) [reply src/components/notifications/dto/notification.dto.ts:32]: Just stubbed. It will be uncommented when implemented & we have edgedb schema types.

**Review by rdonigian — APPROVED** [review]

File: src/components/notifications/dto/notification.dto.ts (hunk 1 above)
Comment (bryanjnelson) [reply src/components/notifications/dto/notification.dto.ts:17]: I'm fine with keeping it as `unread` as well.

**Review by bryanjnelson — APPROVED** [review]

**Pull request merged**